### PR TITLE
Remove canonical URLs from demo metadata

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Title:**
Remove the `canonicalURL` field from demo metadata.

**Summary:**
This PR removes the `canonicalURL` field from existing demo metadata. The `canonicalURL` field has been superseded by the (autogenerated) `slug` field as of #1228.

**Relevant references:**
None.

**Possible Drawbacks:**
None.

**Related GitHub Issues:**
None.

---

**Detailed Changes:**
1. Fixed the `$id` reference in `demo.metadata.schema.0.1.3.json`. 
2. Added `demo.metadata.schema.0.1.4.json` which removes the `canonicalURL` field.
3. Updated the example metadata in `demonstrations_metadata.md`.
4. Removed the `canonicalURL` field from existing demo metadata (and adjusted `dateOfLastModification`).
5. Deleted the obsolete `update_metadata.py` script.

**Merge Checklist:**
* [x] Delete the "Build Website - Text (Test)" GHA workflow  which was added for testing purposes.

**Verification:**

The generated `metadata.json` file in the Glass Onion bucket no longer has any `canonicalURL` fields:

<details><summary><tt>metadata.json</tt> Snippet</summary>
<p>

```json
[
  {
    "authors": [
      {
        "username": "whatsis"
      }
    ],
    "basedOnPapers": [],
    "categories": [
      "Quantum Computing",
      "How-to"
    ],
    "dateOfLastModification": "2024-10-07T00:00:00+00:00",
    "dateOfPublication": "2024-10-01T00:00:00+00:00",
    "doi": "",
    "previewImages": [
      {
        "type": "thumbnail",
        "uri": "/_static/demo_thumbnails/regular_demo_thumbnails/thumbnail_how_to_use_noise_models.png"
      },
      {
        "type": "large_thumbnail",
        "uri": "/_static/demo_thumbnails/large_demo_thumbnails/thumbnail_large_how_to_use_noise_models.png"
      }
    ],
    "referencedByPapers": [],
    "references": [],
    "relatedContent": [
      {
        "id": "tutorial_noisy_circuits",
        "type": "demonstration",
        "weight": 1
      }
    ],
    "seoDescription": "Learn how noise models can be built and inserted into a quantum circuit in PennyLane.",
    "slug": "tutorial_how_to_use_noise_models",
    "tags": [],
    "title": "How to use noise models in PennyLane"
  },
  ...
]
```

</p>
</details>

### CODE REVIEW

**Review by doctorperceptron — APPROVED** [review]
LGTM

File: metadata_schemas/demo.metadata.schema.0.1.3.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "$schema": "https://json-schema.org/draft/2020-12/schema",
-    "$id": "demo.metadata.schema.0.1.2.json",
+    "$id": "demo.metadata.schema.0.1.3.json",
```
Comment (anthayes92) [inline metadata_schemas/demo.metadata.schema.0.1.3.json:3]: ### Question:
Should this be: 
```suggestion
    "$id": "demo.metadata.schema.0.1.4.json",
```
?

**Review by anthayes92 — APPROVED** [review]
Looks good with TODOs :star: One question about version number

File: metadata_schemas/demo.metadata.schema.0.1.3.json (hunk 1 above)
Comment (Mandrenkov) [reply metadata_schemas/demo.metadata.schema.0.1.3.json:3]: Not quite – this is the file from the _previous_ metadata schema!

It was an oversight in #1228 that I'm addressing here.

**Pull request merged**